src/Gui/Inventor/So3DAnnotation.cpp

```diff
@@ -49,21 +55,72 @@ void SoDelayedAnnotationsElement::initClass()
     SO_ENABLE(SoGLRenderAction, SoDelayedAnnotationsElement);
 }
 
-void SoDelayedAnnotationsElement::addDelayedPath(SoState* state, SoPath* path)
+void SoDelayedAnnotationsElement::addDelayedPath(SoState* state, SoPath* path, int priority)
 {
     auto elt = static_cast<SoDelayedAnnotationsElement*>(state->getElementNoPush(classStackIndex));
 
+    // add to priority-aware storage if priority has been specified
+    if (priority > 0) {
+        elt->priorityPaths.emplace_back(path, priority);
+        return;
+    }
+    
     elt->paths.append(path);
 }
 
 SoPathList SoDelayedAnnotationsElement::getDelayedPaths(SoState* state)
 {
     auto elt = static_cast<SoDelayedAnnotationsElement*>(state->getElementNoPush(classStackIndex));
-    auto copy = elt->paths;
-
+    
+    // if we don't have priority paths, just return normal delayed paths
+    if (elt->priorityPaths.empty()) {
+        auto copy = elt->paths;
+        elt->paths.truncate(0);
+        return copy;
+    }
+    
+    // sort by priority (lower numbers render first)
+    std::stable_sort(elt->priorityPaths.begin(), elt->priorityPaths.end(),
+        [](const PriorityPath& a, const PriorityPath& b) {
+            return a.priority < b.priority;
+        });
+    
+    SoPathList sortedPaths;
+    for (const auto& priorityPath : elt->priorityPaths) {
+        sortedPaths.append(priorityPath.path);
+    }
+    
+    // Clear storage
+    elt->priorityPaths.clear();
     elt->paths.truncate(0);
```

Comment:
Is there any difference between `truncate(0)` and `.clear()`? I did copy the truncate one from coin source, but `.clear()` seems like better way. If they are identical it woul'd be good to unify them.